### PR TITLE
Run amazon linux 2 in kmt

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
Adds amazon linux to kernel matrix testing framework.

### Additional Notes
Amazon linux 2 kernel 5.15  is not ready yet.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/system-probe/config/vmconfig.json [hunk 1]
```diff
@@ -8,17 +8,37 @@
                 {
                     "dir": "bionic-server-cloudimg-amd64.qcow2",
                     "tag": "bionic",
-                    "image_source": "https://dd-agent-omnibus.s3.amazonaws.com/kernel-version-testing/rootfs/bionic-server-cloudimg-amd64.qcow2"
+                    "image_source": "https://dd-agent-omnibus.s3.amazonaws.com/kernel-version-testing/rootfs/bionic-server-cloudimg-amd64.qcow2-test-only"
```
Comment (brycekahle) [inline test/new-e2e/system-probe/config/vmconfig.json:11]: What's up with the `-test-only` suffix?

Comment (usamasaqib) [reply test/new-e2e/system-probe/config/vmconfig.json:11]: Comes from [here for example](https://github.com/DataDog/ami-builder/blob/usama.saqib/kernel-version-testing-ami/.gitlab-ci.yml#L139).

I add the suffix when experimenting with different builds, so we don't end up overwriting the stable images used in main.

Here I was experimenting with a leaner build, which removes clang thus reducing the images size by 3Gi.

Comment (brycekahle) [reply test/new-e2e/system-probe/config/vmconfig.json:11]: What is the process for promoting images from `test-only`? Should we merge to `main` references to `test-only` images?

Comment (usamasaqib) [reply test/new-e2e/system-probe/config/vmconfig.json:11]: They should not be merged into `main`. I will remove them once I have the correct images built and pushed to `S3`

**Review by brycekahle — APPROVED** [review]

**Pull request merged**